code/modules/power/tesla/energy_ball.dm

```diff
@@ -128,9 +128,13 @@ var/list/blacklisted_tesla_types = typecacheof(list(/obj/machinery/atmospherics,
 
 /obj/singularity/energy_ball/Bump(atom/A)
 	dust_mobs(A)
+	if(istype(A, /obj/) && !is_type_in_typecache(A, blacklisted_tesla_types))
+		qdel(A)
```

Comment:
This is strictly speaking wrong to do in Bumped(). I'm aware this logic is all over the place and it's not a big deal regardless, but Bump() Cross() Enter() etc are for testing if it's possible and Bumped() Crossed() Entered() etc are for handling what happens when it actually does the action. The reason for this is that Bump() may be called in cases where the two objects don't actually collide, for example because it bumped into another object which stopped movement.

Since I'm not a turbo autist I don't expect you to fix everything or even change this, I'm just spitting it out there so everyone knows and keeps it in mind for future changes because it does annoy me to see people making this mistake over and over.